analytics.go

```diff
@@ -215,28 +210,28 @@ func (c *Client) Track(msg *Track) error {
 
 	msg.Type = "track"
 	c.queue(msg)
-
 	return nil
 }
 
-func (c *Client) startLoop() {
-	go c.loop()
-}
-
 // Queue message.
 func (c *Client) queue(msg message) {
-	c.once.Do(c.startLoop)
 	msg.setMessageId(c.uid())
 	msg.setTimestamp(timestamp(c.now()))
 	c.msgs <- msg
 }
 
 // Close and flush metrics.
-func (c *Client) Close() error {
-	c.quit <- struct{}{}
-	close(c.msgs)
```

Comment:
I moved this to the client's goroutine, it makes the `Close` method simpler and it's easier to understand where the panic could come from.